### PR TITLE
Atmos scaling cvar changes

### PR DESCRIPTION
**Pull request by ElectroJr**
This PR adds a heat-scale argument to `IGasReactionEffect.React()`, with the goal of ensuring that forks/downstreams don't accidentally fail to scale heat outputs, which is an easy mistake to make.

It also adds a non-optional bool to the public `GetHeatCapacity()` method to ensure that external systems intentionally choose to get either the scaled or un-scaled heat capacities. This also changes `TemperatureSystem` to use the un-scaled heat capacities, as it interacts with non-atmos heat capacities that don't get scaled by the atmos cvars.

### CODE REVIEW

File: Content.Server/Atmos/EntitySystems/AtmosphereSystem.cs
Comment (Partmedia) [inline Content.Server/Atmos/EntitySystems/AtmosphereSystem.cs:1]: It doesn't look like there are any relevant changes to this file. Is it okay if I revert this file?

**Review by Partmedia — APPROVED** [review]
Otherwise, looks good, thanks for making these changes. Please do add a note about the breaking changes from the PR template.

File: Content.Server/Atmos/EntitySystems/AtmosphereSystem.cs
Comment (ElectroJr) [reply Content.Server/Atmos/EntitySystems/AtmosphereSystem.cs:1]: The new dependencies are required to remove obsolete code in some of the gas reactions, which I removed while updating all the reactions. Is there a specific reason you want to revert it (i.e. conflicts or something)?

Comment (Partmedia) [reply Content.Server/Atmos/EntitySystems/AtmosphereSystem.cs:1]: Ahh, I see. No, thong specific.

**Pull request merged**